app/safe/internal/server/route/secret/secret.go

```diff
@@ -97,27 +93,42 @@ func Secret(cid string, w http.ResponseWriter, r *http.Request, spiffeid string)
 	}
 
 	log.DebugLn(&cid, "Secret:Upsert: ", "workloadIds:", workloadIds,
-		"namespaces:", namespaces, "backingStore:", backingStore,
+		"namespaces:", namespaces,
 		"template:", template, "format:", format, "encrypt:", encrypt,
 		"appendValue:", appendValue,
 		"notBefore:", notBefore, "expiresAfter:", expiresAfter)
 
 	if len(workloadIds) == 0 && !encrypt {
 		j.Event = event.NoWorkloadId
-		audit.Log(j)
+		journal.Log(j)
 
 		return
 	}
 
 	// `encrypt` means that the value is encrypted, so we need to decrypt it.
 	if encrypt {
-		v, failed := crypto.Decrypt(cid, value, j, w)
-		value = v
+		log.TraceLn(&cid, "Secret: Value is encrypted")
+
+		decrypted, err := crypto.DecryptValue(value)
+
+		// If decryption failed, return an error response.
+		if err != nil {
+			log.InfoLn(&cid, "Secret: Decryption failed", err.Error())
```

Comment:
Inlined the failure response here.

Older implementation was prone to race conditions.